spec/statement_client_spec.rb

```diff
@@ -540,12 +540,22 @@
         client.advance
 
         sleep 1
+
         stub_request(:get, "localhost/v1/next_uri").
           with(headers: headers).
           to_return(body: planning_response.to_json)
+
+        cancelled = false
+        stub_request(:delete, "localhost/v1/next_uri").
```

Comment:
https://github.com/bblimke/webmock?tab=readme-ov-file#setting-expectations-in-rspec-on-the-stub

you can do it with webmock features.

```suggestion
        stub = stub_request(:delete, "localhost/v1/next_uri").

        ...

        # after client.advance
        expect(stub).to have_been_requested
```

Reply:
Oops. Thank you.